### ISSUE TITLE
Fail to scale up cluster via operator REST API

### ISSUE THREAD

**Issue by jhzhu89** [oracle/weblogic-kubernetes-operator#287]
Hi, I followed the instructions - `http(s)://${OPERATOR_ENDPOINT}/operator/v1/domains/domain1/clusters/cluster-1/scale` - to issue the request to scale up the cluster, but only got a 400. 

I modified the script mentioned on [this page ](https://github.com/oracle/weblogic-kubernetes-operator/blob/master/site/rest.md#using-the-operators-rest-services) to POST the request. Here is the command generated by my script, and its output: 
```
curl -k -v --cacert /tmp/operator.cert.pem -H 'Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJ3ZWJsb2dpYy1vcGVyYXRvciIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VjcmV0Lm5hbWUiOiJ3ZWJsb2dpYy1vcGVyYXRvci10b2tlbi03NnF6OSIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50Lm5hbWUiOiJ3ZWJsb2dpYy1vcGVyYXRvciIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50LnVpZCI6IjgwZGM0ZjllLTUzNWUtMTFlOC04ODkyLTBjYzQ3YWIxZjgyNCIsInN1YiI6InN5c3RlbTpzZXJ2aWNlYWNjb3VudDp3ZWJsb2dpYy1vcGVyYXRvcjp3ZWJsb2dpYy1vcGVyYXRvciJ9.AUy2cVBOwLG4uqNyimdu3RmqCYeuj3JvJ02uJf62bcrjD0M_JcQqaoTgdXNqdQohMswoeKaymL26op5289c5JyDr90ZFHXZy2x8O8VSm3_6C_WNYtudFaf6C6sJGAAfSoS_o-_8GqyLlG9_n6empbuaQ2yGq6e9YD0RH6lCvZWpAnHB44g0we5RoEPktrNIMiK8V7gWa-k8h7iaxqjKjtci9pyqKFRDIyVmCdN5pZYi7fqeAVWjfCKL4yGBH3_bUIuJqUIiixdDdbwOY5nNYTMl7VVboul4NPGXIKP2sX3XcJLANAGuJSDVgq7K-fpl969Q4tu_sFKbf7NqUENEygg' -H 'Content-Type: application/json' -X POST https://172.16.1.175:31001/operator/v1/domains/domain2/clusters/cluster-1/scale -d '{"configuredManagedServerCount": 3}'                                                  
* About to connect() to 172.16.1.175 port 31001 (#0)
*   Trying 172.16.1.175...
* Connected to 172.16.1.175 (172.16.1.175) port 31001 (#0)
* Initializing NSS with certpath: sql:/etc/pki/nssdb
* skipping SSL peer certificate verification
* SSL connection using TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
* Server certificate:
*       subject: CN=weblogic-operator
*       start date: 5月 15 07:56:35 2018 GMT
*       expire date: 5月 12 07:56:35 2028 GMT
*       common name: weblogic-operator
*       issuer: CN=weblogic-operator
> POST /operator/v1/domains/domain2/clusters/cluster-1/scale HTTP/1.1
> User-Agent: curl/7.29.0
> Host: 172.16.1.175:31001
> Accept: */*
> Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJ3ZWJsb2dpYy1vcGVyYXRvciIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VjcmV0Lm5hbWUiOiJ3ZWJsb2dpYy1vcGVyYXRvci10b2tlbi03NnF6OSIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50Lm5hbWUiOiJ3ZWJsb2dpYy1vcGVyYXRvciIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50LnVpZCI6IjgwZGM0ZjllLTUzNWUtMTFlOC04ODkyLTBjYzQ3YWIxZjgyNCIsInN1YiI6InN5c3RlbTpzZXJ2aWNlYWNjb3VudDp3ZWJsb2dpYy1vcGVyYXRvcjp3ZWJsb2dpYy1vcGVyYXRvciJ9.AUy2cVBOwLG4uqNyimdu3RmqCYeuj3JvJ02uJf62bcrjD0M_JcQqaoTgdXNqdQohMswoeKaymL26op5289c5JyDr90ZFHXZy2x8O8VSm3_6C_WNYtudFaf6C6sJGAAfSoS_o-_8GqyLlG9_n6empbuaQ2yGq6e9YD0RH6lCvZWpAnHB44g0we5RoEPktrNIMiK8V7gWa-k8h7iaxqjKjtci9pyqKFRDIyVmCdN5pZYi7fqeAVWjfCKL4yGBH3_bUIuJqUIiixdDdbwOY5nNYTMl7VVboul4NPGXIKP2sX3XcJLANAGuJSDVgq7K-fpl969Q4tu_sFKbf7NqUENEygg                                                                                   
> Content-Type: application/json
> Content-Length: 37
>
* upload completely sent off: 37 out of 37 bytes
< HTTP/1.1 400 Bad Request
< Connection: close
< Content-Length: 0
<
* Closing connection 0
```

Could you help explain why it's a bad request, since there was no detailed error message returned? Thanks.

**Comment by moreaut** [comment]
You need to add the an X-Requested-By header to all POST requests against the Operator REST API, including the /scale resouce.  (this protects against cross site scripting attacks).

For example: -H X-Requested-By:MyClient

See https://github.com/oracle/weblogic-kubernetes-operator/blob/master/docs/swagger/index.html
Look for '/scale'.  You'll notice that it lists the X-Requested-By header under its parameters section.

Thanks, -Tom

**Comment by jhzhu89** [comment]
Thank you @moreaut , 

After adding the 'X-Requested-By' header, it worked. BTW, the request body mentioned on [this page](https://github.com/oracle/weblogic-kubernetes-operator/blob/master/site/scaling.md) is `{"configuredManagedServerCount": 3}`

Seems that it should be `{"managedServerCount": 3}`, right? Maybe this page should be upated :)

Thanks,
Jiahao

**Comment by moreaut** [comment]
Thanks Jiahao.  You are correct - it should be 'managedServerCount'.  We'll need to update scaling.md to correct that and to mention the required headers.

**Comment by rjeberhard** [comment]
Fixed with pull request #291
